### PR TITLE
Tests: using GOMOD for the source root to run integration tests outside GOPATH

### PR DESCRIPTION
**Pull request by veggiemonk**
### TLDR

- move `PkSourceRoot` from `pkg/test/world.go` to `internal/osutil/paths.go`;
- migrate all calls from `osutil.GoPackagePath("perkeep.org")` to `osutil.PkSourceRoot()`;
- keep the variable name consistent (`camliRootPath`, `camDir`, etc. => `srcRoot`);
- The GOPATH way is still there as a fallback for compatibility.

Hi, 

While playing around to learn the code base, I tried to run the tests and got errors about `perkeep.org` not being in `GOPATH/src`. 

I thought it would be a good way to get acquainted with the code base. 
I hope I didn't overstep by changing too many things. 
All the tests are passing, including the integration tests.

Let me know what you would like to change in this PR, I'm just trying to be a good citizen. :)

 

For reference: https://go.dev/wiki/GOPATH

### CODE REVIEW

**Comment by bradfitz** [comment]
Ha, I just hit this last night too! You beat me.

I'll take a look later today. (on phone now)

**Comment by bradfitz** [comment]
LGTM, thanks!
